RecoLocalCalo/HcalRecProducers/src/HBHEPhase1Reconstructor.cc

```diff
@@ -121,6 +127,48 @@ namespace {
     {
         return std::pair<bool,bool>(df.linkError(), df.capidError());
     }
+
+    std::unique_ptr<HBHEStatusBitSetter> parse_HBHEStatusBitSetter(
+        const edm::ParameterSet psdigi)
+    {
+        return std::make_unique<HBHEStatusBitSetter>(
+            psdigi.getParameter<double>("nominalPedestal"),
+            psdigi.getParameter<double>("hitEnergyMinimum"),
+            psdigi.getParameter<int>("hitMultiplicityThreshold"),
+            psdigi.getParameter<std::vector<edm::ParameterSet> >("pulseShapeParameterSets"));
+    }
+
+    std::unique_ptr<HBHEPulseShapeFlagSetter> parse_HBHEPulseShapeFlagSetter(
+        const edm::ParameterSet &psPulseShape)
+    {
+        return std::make_unique<HBHEPulseShapeFlagSetter>(
+            psPulseShape.getParameter<double>("MinimumChargeThreshold"),
+            psPulseShape.getParameter<double>("TS4TS5ChargeThreshold"),
```

Comment:
Since you always prefer external references to my direct suggestions:

http://stackoverflow.com/questions/33112338/c-using-struct-arguments-for-functions-instead-of-multiple-arguments
https://en.wikipedia.org/wiki/Law_of_Demeter

http://stackoverflow.com/questions/174968/how-many-parameters-are-too-many

https://books.google.com/books?id=_i6bDeoCQzsC&pg=PA40&dq=The+ideal+number+of+arguments+for+a+function+is+zero&hl=en&sa=X&ved=0ahUKEwi66fH--v_OAhVBSGMKHe_8Bp4Q6AEIHjAA#v=onepage&q=The%20ideal%20number%20of%20arguments%20for%20a%20function%20is%20zero&f=false